### PR TITLE
Fix API for edit and delete release attachment

### PR DESCRIPTION
**Pull request by davidsvantesson**
As title, currently it always return 404.

### CODE REVIEW

File: routers/api/v1/repo/release_attachment.go [hunk 1]
```diff
@@ -242,13 +244,14 @@ func EditReleaseAttachment(ctx *context.APIContext, form api.EditAttachmentOptio
 
 	// Check if release exists an load release
 	releaseID := ctx.ParamsInt64(":id")
-	attachID := ctx.ParamsInt64(":attachment")
+	attachID := ctx.ParamsInt64(":asset")
```
Comment (guillep2k) [inline routers/api/v1/repo/release_attachment.go:247]: I guess you should change:
```
	// - name: attachment_id
	//   in: path
	//   description: id of the attachment to edit
	//   type: integer
	//   format: int64
	//   required: true
```
into
```
	// - name: asset
	//   in: path
	//   description: id of the attachment to edit
	//   type: integer
	//   format: int64
	//   required: true
```

Comment (guillep2k) [reply routers/api/v1/repo/release_attachment.go:247]: Now that I think of it, why `asset`? It should be `attachment_id`, or else the swagger documentation needs update.

**Review by guillep2k — CHANGES_REQUESTED** [review]

File: routers/api/v1/repo/release_attachment.go (hunk 1 above)
Comment (davidsvantesson) [reply routers/api/v1/repo/release_attachment.go:247]: Does swagger and macaron have to match? The GET operation already look like this (and work). There are other APIs where different variables are used in swagger and macaron.

Comment (davidsvantesson) [reply routers/api/v1/repo/release_attachment.go:247]: But sure, this difference is probably why the error was introduced.

Comment (guillep2k) [reply routers/api/v1/repo/release_attachment.go:247]: I guess you're right.

**Review by guillep2k — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Comment by guillep2k** [comment]
Ping LG-TM

**Pull request merged**